R/roles.R

```diff
@@ -171,10 +176,9 @@ add_role <- function(recipe, ..., new_role = "predictor", new_type = NULL) {
   existing_var_idx <- recipe$var_info$variable %in% vars
 
   if (all(is.na(recipe$var_info$role[existing_var_idx]))) {
-    vars <- glue::glue_collapse(glue::single_quote(vars), sep = ", ")
-    rlang::abort(glue(
-      "No role currently exists for column(s): {vars}. Please use ",
-      "`update_role()` instead."
+   cli::cli_abort(c(
+      "!" = "No role currently exists for column{?s}: {.and {.var {vars}}}.",
```

Comment:
```suggestion
      "!" = "No role currently exists for column{?s}: {.var {vars}}.",
```

I believe `{.and {.var {}}}` is redundant:

``` r
# one -------------------------------------------------------------------------
vars <- c("hey", "there")
cli::cli_abort(c(
  "!" = "No role currently exists for column{?s}: {.and {.var {vars}}}.",
  "i" = "Please use {.fn update_role} instead."
))
#> Error:
#> ! No role currently exists for columns: `hey` and `there`.
#> ℹ Please use `update_role()` instead.
#> Backtrace:
#>     ▆
#>  1. └─cli::cli_abort(...)
#>  2.   └─rlang::abort(...)

cli::cli_abort(c(
  "!" = "No role currently exists for column{?s}: {.var {vars}}.",
  "i" = "Please use {.fn update_role} instead."
))
#> Error:
#> ! No role currently exists for columns: `hey` and `there`.
#> ℹ Please use `update_role()` instead.
#> Backtrace:
#>     ▆
#>  1. └─cli::cli_abort(...)
#>  2.   └─rlang::abort(...)

# two -------------------------------------------------------------------------
vars <- c("hey")
cli::cli_abort(c(
  "!" = "No role currently exists for column{?s}: {.and {.var {vars}}}.",
  "i" = "Please use {.fn update_role} instead."
))
#> Error:
#> ! No role currently exists for column: `hey`.
#> ℹ Please use `update_role()` instead.
#> Backtrace:
#>     ▆
#>  1. └─cli::cli_abort(...)
#>  2.   └─rlang::abort(...)

cli::cli_abort(c(
  "!" = "No role currently exists for column{?s}: {.var {vars}}.",
  "i" = "Please use {.fn update_role} instead."
))
#> Error:
#> ! No role currently exists for column: `hey`.
#> ℹ Please use `update_role()` instead.
#> Backtrace:
#>     ▆
#>  1. └─cli::cli_abort(...)
#>  2.   └─rlang::abort(...)

# three ------------------------------------------------------------------------
vars <- c("hey", "there", "friend")
cli::cli_abort(c(
  "!" = "No role currently exists for column{?s}: {.and {.var {vars}}}.",
  "i" = "Please use {.fn update_role} instead."
))
#> Error:
#> ! No role currently exists for columns: `hey`, `there`, and `friend`.
#> ℹ Please use `update_role()` instead.
#> Backtrace:
#>     ▆
#>  1. └─cli::cli_abort(...)
#>  2.   └─rlang::abort(...)

cli::cli_abort(c(
  "!" = "No role currently exists for column{?s}: {.var {vars}}.",
  "i" = "Please use {.fn update_role} instead."
))
#> Error:
#> ! No role currently exists for columns: `hey`, `there`, and `friend`.
#> ℹ Please use `update_role()` instead.
#> Backtrace:
#>     ▆
#>  1. └─cli::cli_abort(...)
#>  2.   └─rlang::abort(...)
```

<sup>Created on 2023-10-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

This change applies to the "Role ... already exists for..." message, as well.